torch/csrc/jit/codegen/cuda/shape_inference.cpp

```diff
@@ -492,18 +574,31 @@ class NaiveTypePropagator {
           ? *scalar_type
           : c10::promoteTypes(*op0->scalarType(), *op1->scalarType());
 
-      return TensorType::create(
-          promoted_scalar_type, *op0->device(), c10::nullopt, c10::nullopt);
+      if (op0->isComplete() && op1->isComplete()) {
```

Comment:
nitpick `isComplete`

Reply:
`getBroadcastSizes` is not used anywhere else, maybe we can update the return value as an optional?